### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by xavi-**
Fixed the following issue:
- Added code to populate signal with proper hostname
- Accidentally hard coded time that gets parsed
- Removed `offset + 1` adjustment

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -42,11 +45,40 @@ func sendToSignalFX(delay int64) error {
...
+		Dial: func(network, addr string) (net.Conn, error) {
+			return net.DialTimeout(network, addr, time.Duration(3*time.Second))
+		},
+	}
+
+	client := &http.Client{Transport: transport}
+
+	res, err := client.Get("http://169.254.169.254/latest/meta-data/local-ipv4")
+	if err != nil {
+		log.Error("meta-data-request-failed")
```
Comment (nathanleiby) [inline main.go]: Should we log the specific error that occurred?

File: main.go [hunk 2]
```diff
@@ -42,11 +45,40 @@ func sendToSignalFX(delay int64) error {
...
+		return "unknown ip"
+	}
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Error("meta-data-parse-failed")
```
Comment (nathanleiby) [inline main.go]: Should we log the specific error that occurred?

File: main.go [hunk 3]
```diff
@@ -194,7 +194,7 @@ func trackTimestamp(posFile string) (time.Time, string, error) {
 		return time.Time{}, "", fmt.Errorf("No timestamp found")
 	}
 
-	ts, err := time.Parse(time.RFC3339Nano, "2017-05-21T22:49:23.314299+00:00")
+	ts, err := time.Parse(time.RFC3339Nano, line[:32])
```
Comment (nathanleiby) [inline main.go:196]: Is it OK if parsing fails occasionally? I think we may still have 2 timestamp formats in the wild (although ideally everything should be this new format!)

https://github.com/Clever/kinesis-to-firehose/blob/master/decode/decode_test.go#L96-L117

**Review by nathanleiby — APPROVED** [review]
just a few nits/questions. :shipit:

File: main.go (hunk 3 above)
Comment (xavi-) [reply main.go:196]: Oh, I thought everything was converted to nano second

File: main.go (hunk 1 above)
Comment (xavi-) [reply main.go]: Updated in e1e0822c270eb39d9e863218940570bf4ad645e9

File: main.go (hunk 2 above)
Comment (xavi-) [reply main.go]: Updated in e1e0822c270eb39d9e863218940570bf4ad645e9

File: main.go (hunk 3 above)
Comment (xavi-) [reply main.go:196]: Good catch, would be okay to use [`time.Stamp`](https://golang.org/pkg/time/#pkg-constants)?

Comment (xavi-) [reply main.go:196]: Added fallback in 7b546320db628cad31f7d339ce82e97f3d116883

**Pull request merged**